machine/topology_test.go

```diff
@@ -25,6 +25,88 @@ import (
 	"github.com/google/cadvisor/utils/sysfs/fakesysfs"
 )
 
+func TestPhysicalCores(t *testing.T) {
+	testfile := "./testdata/cpuinfo"
+	testcpuinfo, err := ioutil.ReadFile(testfile)
+	if err != nil {
+		t.Fatalf("unable to read input test file %s", testfile)
+	}
+
+	numPhysicalCores := GetPhysicalCores(testcpuinfo)
+	if numPhysicalCores != 6 {
```

Comment:
`stretchr/testify` is cAdvisor dependency and it introduces very neat [`assert`](https://godoc.org/github.com/stretchr/testify/assert) package :) Consider using it instead of `if`s.